src/hyperion/external_interaction/callbacks/xray_centre/callback_collection.py

```diff
@@ -24,15 +29,17 @@
 class XrayCentreCallbackCollection(AbstractPlanCallbackCollection):
     """Groups the callbacks for external interactions in the fast grid scan, and
     connects the Zocalo and ISPyB handlers. Cast to a list to pass it to
-    Bluesky.preprocessors.subs_decorator()."""
+    Bluesky.preprocessors.subs_decorator().
+
+    Note: currently specific to i03 as it creates i03's goniometer axes."""
 
     nexus_handler: GridscanNexusFileCallback
     ispyb_handler: GridscanISPyBCallback
     zocalo_handler: XrayCentreZocaloCallback
 
     @classmethod
     def from_params(cls, parameters: InternalParameters):
-        nexus_handler = GridscanNexusFileCallback()
+        nexus_handler = GridscanNexusFileCallback(create_i03_goniometer_axes)
```

Comment:
Still feels like we should be passing this from "higher up" the call chain. But probably fine for now. Possibly create an issue to think about where this comes from.